src/blocked.c

```diff
@@ -251,6 +251,7 @@ void unblockClient(client *c, int queue_for_reprocessing) {
     c->bstate->btype = BLOCKED_NONE;
     c->bstate->unblock_on_nokey = 0;
     removeClientFromTimeoutTable(c);
+    c->bstate->timeout = 0;
```

Comment:
This seems to be breaking some other tests.

Reply:
Looks like it needs to be reset on creation instead, since query reprocessing relies on the original timeout value. Should be fixed now